### PR TITLE
added release workflow

### PR DESCRIPTION
**Pull request by danabens**
added release workflow triggered from manually creating a release in github

note that test credentials and test repo are used until further review

passing in fork: https://github.com/danabens/sagemaker-experiments/commit/f05b09d8e23fe932f98383f774650013afeaf7ce/checks?check_suite_id=400665622

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 
-on: pull_request
+on:
+  pull_request:
+  release:
+    types: [published]
 
 jobs:
   build:
     runs-on: ubuntu-latest
```
Comment (l2yao) [inline .github/workflows/main.yml]: I think we might want to run on Amazon Linux as well, but this can be done later

File: tests/conftest.py [hunk 2]
```diff
@@ -108,11 +111,13 @@ def trials(experiment_obj, sagemaker_boto_client):
...
-        experiment.Experiment.create(experiment_name=experiment_name,
-                                     sagemaker_boto_client=sagemaker_boto_client)
-        for experiment_name in names()
-    ]
+    experiment_objs = []
+
+    for experiment_name in names():
+        experiment_objs.append(experiment.Experiment.create(experiment_name=experiment_name,
+                                     sagemaker_boto_client=sagemaker_boto_client))
+        time.sleep(1)
```
Comment (l2yao) [inline tests/conftest.py]: whats the reason for sleeping here? is there a async process here?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,11 +1,13 @@
 name: Main
 
-on: pull_request
+on:
+  pull_request:
```
Comment (l2yao) [inline .github/workflows/main.yml]: Lets enforce it to master branch

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -1,11 +1,13 @@
 name: Main
```
Comment (l2yao) [inline .github/workflows/main.yml]: do we want a more meaningful name for the workflow here?

**Comment by l2yao** [comment]
Also I see there is a conflicts detected, can you resolve that as well, Thanks

File: .github/workflows/main.yml (hunk 1 above)
Comment (danabens) [reply .github/workflows/main.yml]: ya maybe in the future, github makes it easy build and test on various platforms (mac and windows) and we could use a self hosted runner for amazon linux.

File: tests/conftest.py (hunk 2 above)
Comment (danabens) [reply tests/conftest.py]: there was the problem of throttling of requests causing test failures.  I've already bumped up the limits for the sdk account but i wouldnt be surprised if we encountered throttling issue with multiple developers X multiple concurrent test runs (python 3.5, 3.6, 3.7)

also if you look at other tests there are already a lot of sleeps.

for now tests are not hitting throttling issues though.

File: .github/workflows/main.yml (hunk 3 above)
Comment (danabens) [reply .github/workflows/main.yml]: added

File: .github/workflows/main.yml (hunk 4 above)
Comment (danabens) [reply .github/workflows/main.yml]: Updated

File: .github/workflows/build_release.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+        echo "${{ secrets.PYPI_SIGN_PRIVATE_KEY }}" | gpg --batch --import --no-default-keyring --keyring ./sessionring.gpg
+        gpg --no-default-keyring --keyring ./sessionring.gpg --pinentry-mode loopback --passphrase ""${{ secrets.PYPI_SIGN_PASSPHRASE }}"" --detach-sign -ao /dev/null dist/*
+    - name: Twine Check
+      run: twine check dist/*
+    - name: Publish to PyPi
+      uses: pypa/gh-action-pypi-publish@master
+      with:
+        user: ${{ secrets.PYPI_USERNAME }}
+        password: ${{ secrets.PYPI_PASSWORD }}
+        repository_url: https://test.pypi.org/legacy/
```
Comment (owen-t) [inline .github/workflows/build_release.yml]: nit - add a newline

**Pull request merged**